vdp/pipeline/v1beta/integration.proto

```diff
@@ -192,6 +192,26 @@ message Integration {
     string url = 2 [(google.api.field_behavior) = OUTPUT_ONLY];
   }
 
+  // OAuthConfig contains the configuration parameters for fetching an access
+  // token via an OAuth 2.0 flow.
+  message OAuthConfig {
+    // The URL of the OAuth server to initiate the authentication and
+    // authorization process.
+    string auth_url = 1 [
+      (validate.rules).string.uri = true,
+      (google.api.field_behavior) = OUTPUT_ONLY
+    ];
+    // The URL of the OAuth server to exchange the authorization code for an
+    // access token.
+    string access_url = 2 [
+      (validate.rules).string.uri = true,
+      (google.api.field_behavior) = OUTPUT_ONLY
+    ];
+    // A list of scopes that identify the resources that the connection will be
+    // able to access on the user's behalf.
+    repeated string scopes = 11 [(google.api.field_behavior) = OUTPUT_ONLY];
```

Comment:
Other fields needed for the OAuth flow, perhaps the client must hold these values rather than the integration:
- `client_id`: will change based on the environment.
- `client_secret`: will change based on the environment. Should this even be passed to the client?
- `redirect_uri`: will change based on the environment.
- `state`: for security and robustness. Can be defined in the client.
- `prompt`: can be defined by the client.
- `response_type`: fixed for Google OAuth, must be `"code"`.
- `grant_type`: fixed for Google OAuth, must be `"authorization_code"`.

Reply:
🤝 agreed to pass Client ID and Secret to console via env variables. The rest of the fields have to do with the client-vendor negotiation and don't need to change on a per-component basis.